synfig-core/src/synfig/os.h

```diff
@@ -0,0 +1,101 @@
+/* === S Y N F I G ========================================================= */
+/*!	\file os.h
+**	\brief OS specific methods
+**
+**	\legal
+**	Copyright (c) 2002-2005 Robert B. Quattlebaum Jr., Adrian Bentley
+**	......... ... 2021 Rodolfo Ribeiro Gomes
+**
+**	This file is part of Synfig.
+**
+**	Synfig is free software: you can redistribute it and/or modify
+**	it under the terms of the GNU General Public License as published by
+**	the Free Software Foundation, either version 2 of the License, or
+**	(at your option) any later version.
+**
+**	Synfig is distributed in the hope that it will be useful,
+**	but WITHOUT ANY WARRANTY; without even the implied warranty of
+**	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+**	GNU General Public License for more details.
+**
+**	You should have received a copy of the GNU General Public License
+**	along with Synfig.  If not, see <https://www.gnu.org/licenses/>.
+**	\endlegal
+*/
+/* ========================================================================= */
+
+#ifndef SYNFIG_OS_H
+#define SYNFIG_OS_H
+
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace synfig {
+
+namespace OS {
+
+class RunArgs
+{
+	std::vector<std::string> run_args;
+public:
+	void push(const std::vector<std::string>& args);
+	void push(const std::vector<const char*>& args);
+	void push(const std::string& item);
+	void push(std::string&& item);
+	void push(const char *__format, ...);
+	void push_pair(const std::string&  key, const std::string&  value);
+	void push_pair_filename(const std::string&  key, const std::string&  filename);
+	void push_filename(const std::string& filename);
+	void push_filename(std::string&& filename);
+	void push_filename(const char *filename);
+	void clear();
+
+	std::string get_string() const;
+	const std::string& operator[](size_t i) const;
+	size_t size() const;
+	bool empty() const;
+};
+
+enum RunMode {
+	RUN_MODE_NONE,
+	RUN_MODE_READ,
+	RUN_MODE_WRITE,
+	RUN_MODE_READWRITE,
+};
+
+struct RunPipe {
+	typedef std::unique_ptr<RunPipe> Handle;
+	virtual ~RunPipe() = default;
+	virtual bool init(std::string binary_path, const RunArgs& binary_args, RunMode mode, const std::string& stdout_redir_file = "", const std::string& stdin_redir_file = "") = 0;
```

Comment:
I think it's better to rename it to `open(...)` to match the `close()` method.

Reply:
Indeed.